mne/stats/regression.py

```diff
@@ -134,3 +143,208 @@ def _fit_lm(data, design_matrix, names):
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (1.) is
+        used.
+    covariates : dict-like | None
+        If dict-like, values have to be array-like and of the same length as
+        the columns in ```events```. Keys correspond to additional event
+        types/conditions to be estimated and are matched with the time points
+        given by the first column of ```events```. If None, only binary events
+        (from event_id) are used.
+    reject : bool | dict
+        Activate rejection parameters based on peak-to-peak amplitude in
+        continuously selected subwindows. If reject is None, no rejection is
+        done. If True, the following default is employed:
+            reject = dict(grad=4000e-12, # T / m (gradiometers)
+                          mag=4e-11, # T (magnetometers)
+                          eeg=40e-5, # uV (EEG channels)
+                          eog=250e-5 # uV (EOG channels)
+                         )
+        If dict, keys are types ('grad' | 'mag' | 'eeg' | 'eog' | 'ecg') and
+        values are the maximal peak-to-peak values to select rejected epochs.
+    tstep : float
+        Length of windows for peak-to-peak detection.
+    decim : int
+        Decimate by choosing only a subsample of data points. Highly
+        recommended for data recorded at high sampling frequencies, as
+        otherwise huge intermediate matrices have to be created and inverted.
+    picks : None | list
+        List of indices of channels to be included. If None, defaults to all
+        MEG and EEG channels.
+    solver : str | function
+        Either a function which takes as its inputs the predictor matrix X
+        and the observation matrix Y, and returns the coefficient matrix b;
+        or a string (for now, only 'defaults'), in which case the solver used
+        is dot(scipy.linalg.pinv(dot(X.T, X)), dot(X.T, Y.T)).T.
+
+    Returns
+    -------
+    ev_dict : dictionary
+        A dictionary where the keys correspond to conditions and the values are
+        Evoked objects with the rE[R/F]Ps. These can be used exactly like any
+        other Evoked object, including e.g. plotting or statistics.
+    """
+
+    if solver == 'default':
+        fast_dot = get_fast_dot()
+
+        # inv is slightly (~10%) faster, but pinv seemingly more stable
+        def solver(X, Y):
+            return fast_dot(linalg.pinv(fast_dot(X.T, X)),
+                            fast_dot(X.T, Y.T)).T
+
+    # prepare raw and events
+    if picks is None:
+        picks = pick_types(raw.info, meg=True, eeg=True, ref_meg=True,
+                           stim=False, eog=False, ecg=False,
+                           emg=False, exclude=['bads'])
+    raw = raw.pick_channels([raw.ch_names[ii] for ii in picks], copy=True)
```

Comment:
We need the info later for creating the evokeds. This seemed the most straight-forward approach to make sure everything is in sync. 